reframe/utility/osext.py

```diff
@@ -415,10 +415,11 @@ def is_url(s):
     return parsed.scheme != '' and parsed.netloc != ''
 
 
-def git_clone(url, targetdir=None, timeout=5):
+def git_clone(url, targetdir=None, opts=[], timeout=5):
```

Comment:
Do not use `[]` as a default argument because it's mutable. Use `None` and then set the `opts` internally as:

```python
opts = opts if opts is not None else []
```